semgrep/semgrep/meta.py

```diff
@@ -215,25 +208,89 @@ def commit_sha(self) -> Optional[str]:
             return os.getenv("GITHUB_SHA")
         return super().commit_sha
 
-    @property
-    def head_ref(self) -> Optional[str]:
-        if self.is_pull_request_event:
-            return self.commit_sha
-        else:
-            return None
+    def _shallow_fetch_branch(self, branch_name: str) -> None:
+        """
+        Split out shallow fetch so we can mock it away in tests
+        """
+        subprocess.run(
+            [
+                "git",
+                "fetch",
+                "origin",
+                "--depth",
+                "1",
+                "-fu",
+                f"{branch_name}:{branch_name}",
+            ],
+            check=True,
+            capture_output=True,
+            encoding="utf-8",
+            timeout=GIT_SH_TIMEOUT,
+        )
 
-    @property
-    def base_branch_tip(self) -> Optional[str]:
-        return self.glom_event(T["pull_request"]["base"]["sha"])  # type: ignore
+    def _get_latest_commit_hash_in_branch(self, branch_name: str) -> str:
+        """
+        Return sha hash of latest commit in a given branch
+
+        Does a git fetch of given branch with depth = 1
+        """
+        # Fetch latest
+        self._shallow_fetch_branch(branch_name)
+        branch_rev_parse = subprocess.run(
+            ["git", "rev-parse", branch_name],
+            encoding="utf-8",
+            check=True,
+            timeout=GIT_SH_TIMEOUT,
+            capture_output=True,
+        ).stdout.rstrip()
```

Comment:
```suggestion
        ).stdout.strip()
```

if this works as well, let's use `strip()` like elsewhere in this file, otherwise we'll forever be wondering why `rstrip` is needed for this one method 😅